src/render/opengl/SDL_render_gl.c

```diff
@@ -885,20 +885,42 @@ static int GL_SetRenderTarget(SDL_Renderer *renderer, SDL_Texture *texture)
 
     texturedata = (GL_TextureData *)texture->driverdata;
     data->glBindFramebufferEXT(GL_FRAMEBUFFER_EXT, texturedata->fbo->FBO);
-    /* TODO: check if texture pixel format allows this operation */
     data->glFramebufferTexture2DEXT(GL_FRAMEBUFFER_EXT, GL_COLOR_ATTACHMENT0_EXT, data->textype, texturedata->texture, 0);
+
     /* Check FBO status */
+    /* cf. https://registry.khronos.org/OpenGL-Refpages/gl4/html/glCheckFramebufferStatus.xhtml */
     status = data->glCheckFramebufferStatusEXT(GL_FRAMEBUFFER_EXT);
     if (status != GL_FRAMEBUFFER_COMPLETE_EXT) {
-        if ((err = data->glGetError()) != GL_NO_ERROR) {
-            switch (err) {
+        if ((glErr = data->glGetError()) == GL_NO_ERROR) {
+            return 0;
```

Comment:
This is still returning success when we get an incomplete framebuffer. I think there was a misunderstanding about our previous comments, where the flow of logic could accidentally fail to return an error in this case, now it's checking for that case explicitly and not returning an error.